zerver/lib/test_classes.py

```diff
@@ -2393,6 +2395,13 @@ def setUp(self) -> None:
             last_updated=timezone_now(),
         )
         self.server.save()
+
+        # Realms should have matching RemoteRealm entries - simulating having realms registered
+        # with the bouncer, which is going to be the primary case for modern servers. Tests
+        # wanting to have missing registrations, or simulating legacy server scenarios,
+        # should delete RemoteRealms to explicit set things up.
+        update_remote_realm_data_for_server(self.server, get_realms_info_for_push_bouncer())
```

Comment:
Do you think it might be better to move this into `populate_db`, so that we have existing registrations in the development environment database too? Would cut a step for this sort of testing.

Reply:
@timabbott Yeah I think that would be significantly nicer in principle, but this is `zilencer` logic (`RemoteRealm`) and we currently don't put any `zilencer` logic in `populate_db`. We do in `populate_analytics_db`, but that only runs `if not options["test_suite"]:`.

I think this is to let tests work without zilencer, right? Same reason we put `@skipUnless(settings.ZILENCER_ENABLED, "requires zilencer")` around some tests.

I guess that's also why we create `RemoteZulipServer`s in `BouncerTestCase` instead of `populate_db`.

Reply:
There's no reason to not put `zilencer` stuff in `populate_db`; `populate_db` itself is in zilencer. 

The main purpose of that being a separate app is just to not have it included in production systems that don't need it; a few commands are dangerous to run if you don't know what you're doing (like `populate_db` can wipe your database) .

Reply:
(Probably that `skipUnless` code is just stale; I don't think it makes any sense to have tests conditional on that -- though of course some might want to override that setting to test something). `ZILENCER_ENABLED` is always true in the test suite...

Reply:
@timabbott Hmm, I just remember this was added in 2020 in 468c5b9a58b613d88dcb100ab2e5e32de90e579f and back then `populate_db` was also in `zilencer`. But I don't know the context for that commit and maybe indeed this is just stale now?

So I pushed that first commit to just add this `RemoteRealm` setup in `populate_db`. And in the spirit of the assumption of that `@skipunless` etc. code being stale, pushed an extra commit deleting these conditionals in tests.